carma-messenger-core/cpp_message/src/cpp_message.cpp

```diff
@@ -1244,16 +1287,77 @@ namespace cpp_message
         std::vector<uint8_t> b_array(array_length);
         for(auto i = 0; i < array_length; i++) b_array[i] = buffer[i];
         // for(auto i = 0; i < array_length; i++) std::cout<< (int)b_array[i]<< ", ";
+
+        free(message);
+        free(output_v01_testtest);
+
+        // TCM body
+        free(output_v01);
+        free(output_64b);
+        free(output_128b);
+
+        // TCM package
+        free(output_package);
+        free(label_p);
+        free(tcids);
+        free(label_content);
+        for ( int i = 0; i < tcids_len; i++ )
+        {
+            free(tcid_output_128b[i]);
+            free(tcid_val[i]);
+        }
+        free(tcid_output_128b);
+        free(tcid_val);
+
+        // TCM params
+        free(output_params);
+        free(vclasses_list);
+        free(output_schedule);
+        free(start_p);
+        free(end_p);
+        free(between_list);
+        free(dow_output);
+        free(dow_val);
+        free(vclass_output);
+        free(schedule_output);
+        free(repeat_output);
+        free(detail_output);
+        free(signal_content);
+        for(auto i = 0; i < latperm_size; i++)
+        {
+            free(latperm_items[i]);
+        }
+        free(latperm_items);
+        free(latperm_p);
+        
+        // TCM geometry
+        free(output_geometry);
+        free(proj_content);
+        free(datum_content);
+        free(reftime_val);
+        free(reftime_p);
+        free(nodes_list);
+        for ( int i = 0; i < nodes_len; i++ )
+        {
+            free(pathnode_output[i]);
+            free(z_temp[i]);
+            free(width_temp[i]);
+        }
+        free(pathnode_output);
+        free(z_temp);
+        free(width_temp);
         return boost::optional<std::vector<uint8_t>>(b_array);
     }
 
+    // Classic memory leaks
```

Comment:
Not sure what this comment is saying

Reply:
removed. Just me being exasperated at the functions below.